packages/react-ui-core/src/DropDownMenu/ItemWrapper.js

```diff
@@ -0,0 +1,52 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import cn from 'classnames'
+import themed from 'react-themed'
+
+@themed(/^DrownDownMenu/, { pure: true })
+
+export default class ItemWrapper extends Component {
+  static propTypes = {
+    children: PropTypes.node,
+    index: PropTypes.number,
+    selectedIndex: PropTypes.number,
+    theme: PropTypes.object,
+    onSelectionHover: PropTypes.func,
+    onSelectionClick: PropTypes.func,
+  }
+
+  constructor(props) {
+    super(props)
+    this.handleHover = this.handleHover.bind(this)
+  }
+
+  handleHover() {
+    this.props.onSelectionHover(this.props.index)
+  }
+
+  render() {
+    const {
+      selectedIndex,
+      index,
+      theme,
+      children,
+      onSelectionClick,
+      ...props
+    } = this.props
+
+    return (
+      <div
+        data-tid="dropdown-menu-list-item"
+        className={cn(
+          theme.DropDownMenu_ListItem,
+          selectedIndex === index && theme.DropDownMenu_Selected
+        )}
+        onClick={onSelectionClick}
```

Comment:
We will most likely have an issue here.  If someone passes in `onClick` it will override this based on the spreading of props further down.  Suggestions:
* disallow `onClick` prop from being spread
* rename `onSelectionClick` to `onClick` (this would be my choice 😄 )

Reply:
Gotcha. I'll go with your suggestion.

Reply:
Whatever works, more of a suggestion.  Either way though we'd have to fix something 👍

Reply:
So, I think we didn't call the prop `onClick` because we didn't want to have an `onClick` event handler for the entire `List`. So I'll go with destructuring `onClick` in render to make sure it doesn't override ours.